src/containerapp/azext_containerapp/custom.py

```diff
@@ -4297,3 +4306,195 @@ def delete_workload_profile(cmd, resource_group_name, env_name, workload_profile
+        return
+    user_input = input("Do you want to apply all the patches or specify by id? (y/n/id)\n")
+    patch_apply(cmd, patchable_check_results, user_input, pack_exec_path)
+
+
+def patch_apply(cmd, resource_group_name=None, managed_env=None, show_all=False):
+    if is_docker_running() is False:
+        logger.warning("Please install or start Docker and try again.")
+        return
+    patchable_check_results = patch_list(cmd, resource_group_name, managed_env, show_all=show_all)
+    pack_exec_path = get_pack_exec_path()
+    if patchable_check_results is None:
+        return
+    patchable_check_results_json = json.dumps(patchable_check_results, indent=2)
+    without_unpatchable_results = []
+    without_unpatchable_results = [result for result in patchable_check_results if result["id"] is not None]
+    if without_unpatchable_results == [] and (patchable_check_results is None or show_all is False):
+        return
+    print(patchable_check_results_json)
+    if without_unpatchable_results == []:
+        return
+    patch_apply(cmd, patchable_check_results, "y", pack_exec_path)
+
+
+def patch_apply(cmd, patch_check_list, method, pack_exec_path):
+    m = method.strip().lower()
+    # Track number of times patches were applied successfully.
+    patch_apply_count = 0
+    if m == "y":
+        for patch_check in patch_check_list:
+            if patch_check["id"]:
+                if patch_check["newRunImage"]:
+                    patch_cli_call(cmd,
+                                   patch_check["targetResourceGroup"],
+                                   patch_check["targetContainerAppName"],
+                                   patch_check["targetContainerName"],
+                                   patch_check["targetImageName"],
+                                   patch_check["newRunImage"],
+                                   pack_exec_path)
+                    # Increment patch_apply_count with every successful patch.
+                    patch_apply_count += 1
+    elif m == "n":
+        print("No patch applied.")
+        return
+    else:
+        # Check if method is an existing id in the list
+        for patch_check in patch_check_list:
+            if patch_check["id"] == method:
+                patch_cli_call(cmd,
+                               patch_check["targetResourceGroup"],
+                               patch_check["targetContainerAppName"],
+                               patch_check["targetContainerName"],
+                               patch_check["targetImageName"],
+                               patch_check["newRunImage"],
+                               pack_exec_path)
+                patch_apply_properties = {
+                    'Context.Default.AzureCLI.PatchApplyUserResponse': method,
+                    'Context.Default.AzureCLI.PatchApplyCount': 1
+                }
+                telemetry_core.add_extension_event('containerapp', patch_apply_properties)
```

Comment:
@snehapar9 @harryli0108 -- can we remove this telemetry call and defer to the one at the end of the function? We would need to ensure that this flow doesn't hit the `return` statement below (and we should also add an `else` below for the case where the provided input is not a valid patch method or ID)

Reply:
I believe this return statement was added here because there is no support for passing multiple Ids as arguments at this time. So, if the id provided by the user matches the id in the ACA list from the `patch_list` method, patch is applied once and control returns to the calling function. 
I agree that adding an if-else block here would improve readability. We would also be able to avoid making this telemetry call twice.

Reply:
Updated: changed the telemetry recording method and reduced the return statement so that it'll all end at the last return.